src/app/SubscriptionResumptionHelper.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/InteractionModelEngine.h>
+#include <app/SubscriptionResumptionHelper.h>
+
+namespace chip {
+namespace app {
+SubscriptionResumptionHelper::SubscriptionResumptionHelper() :
+    mOnConnectedCallback(HandleDeviceConnected, this), mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+{}
+
+CHIP_ERROR SubscriptionResumptionHelper::ResumeSubscription(CASESessionManager & caseSessionManager,
+                                                            SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo)
+{
+    mNodeId         = subscriptionInfo.mNodeId;
+    mFabricIndex    = subscriptionInfo.mFabricIndex;
+    mSubscriptionId = subscriptionInfo.mSubscriptionId;
+    mMinInterval    = subscriptionInfo.mMinInterval;
+    mMaxInterval    = subscriptionInfo.mMaxInterval;
+    mFabricFiltered = subscriptionInfo.mFabricFiltered;
+    mAttributePaths = std::move(subscriptionInfo.mAttributePaths);
+    mEventPaths     = std::move(subscriptionInfo.mEventPaths);
+
+    ScopedNodeId peerNode = ScopedNodeId(mNodeId, mFabricIndex);
+    caseSessionManager.FindOrEstablishSession(peerNode, &mOnConnectedCallback, &mOnConnectionFailureCallback);
+    return CHIP_NO_ERROR;
+}
+
+void SubscriptionResumptionHelper::HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr,
+                                                         const SessionHandle & sessionHandle)
+{
+    Platform::UniquePtr<SubscriptionResumptionHelper> _this(static_cast<SubscriptionResumptionHelper *>(context));
```

Comment:
how do you know that this was allocated on the heap when it was set up in the constructor?

Reply:
Changed it